### DISCUSSION TITLE
Mac Book Pro，Apple芯片 编译失败

### DISCUSSION THREAD

**Discussion by maxfong** [alibaba/xquic#448] [Q&A] [answered]
最后一步 make -j

报错：
ld: warning: ignoring file '/usr/local/lib/libevent-2.1.7.dylib': found architecture 'arm64', required architecture 'x86_64'

Apple 芯片的 libevent 默认 arm64 的

**Comment by fw8899** [comment, marked as answer]
是使用homebrew安装的libevent吗？
如果shell是x86_64（2023年之前大部分shell只有x86_86的homebrew，通过Rosetta转译运行），默认安装的lib也是x86_64的，和clang的编译目标arm64不匹配。可以在shell中使用`arch`命令来确认。

处理方法：
1. 删除x86_64的libevent，`brew uninstall --ignore-dependencies libevent`。
2. 删除重新安装arm64版本的shell（比如最新版本的iTerm2）后，通过`brew install libevent`来安装。使用objdump等工具确认lib是否为arm64。
3. 也可以用默认参数自行build libevent并install，或者使用vcpkg之类的包管理器作为依赖添加。

**Comment by fw8899** [comment]
openssl也会遇到一样的问题，处理方法相同。
